lib/src/client/http2_connection.dart

```diff
@@ -77,16 +78,28 @@ class Http2ClientConnection implements connection.ClientConnection {
 
   Future<Socket> _createSocket() async {
     final securityContext = credentials.securityContext;
+    dynamic _host = host;
+    int _port = port;
+    if (host != null) {
+      // `host` which start with 'unix://' prefix, will be parsed as unix socket.
+      RegExp re = RegExp(r'(^unix://)');
+      Match m = re.firstMatch(host);
```

Comment:
Also assumes that `host` is a string. See comments in server.dart, and make this:
```dart
if (host is String && host.startsWith("unix://")) {
```